cpp/src/arrow/array_dict_test.cc

```diff
@@ -1068,17 +1068,17 @@ TEST(TestDictionary, DISABLED_ListOfDictionary) {
 
   ASSERT_OK(list_builder->Append());
   std::vector<std::string> expected;
-  for (char a : "abc") {
-    for (char d : "def") {
-      for (char g : "ghi") {
-        for (char j : "jkl") {
-          for (char m : "mno") {
-            for (char p : "pqr") {
+  for (char a : util::string_view("abc")) {
+    for (char d : util::string_view("def")) {
+      for (char g : util::string_view("ghi")) {
+        for (char j : util::string_view("jkl")) {
+          for (char m : util::string_view("mno")) {
+            for (char p : util::string_view("pqr")) {
```

Comment:
Curious if this was necessary

Reply:
https://github.com/apache/arrow/pull/4930#discussion_r306854941